manifests/packages.pp

```diff
@@ -2,14 +2,37 @@
   if($::osfamily == 'debian') {
     include apt
 
-    apt::pin {'wireguard':
-      packages => ['wireguard-dkms', 'wireguard-tools'],
-      release  => 'experimental',
-      priority => 501,
-      require  => Apt::Source['debian_unstable'],
+    $osname = downcase($facts['os']['name'])
+
+    case $osname {
+      'ubuntu' : {
+        apt::source { "wireguard-$::lsbdistcodename" :
+          location => "http://ppa.launchpad.net/wireguard/wireguard/$osname",
+          release  => $::lsbdistcodename,
+          repos    => 'main',
+          key      => {
+            'id'     => 'E1B39B6EF6DDB96564797591AE33835F504A1A25',
+            'server' => 'pgp.mit.edu',
+          },
+          include  => {
+            'src' => false,
+          },
+        }
+        [Apt::Source["wireguard-$::lsbdistcodename"], Class['apt::update']] -> Package<| title == 'wireguard-tools' |>
+        [Apt::Source["wireguard-$::lsbdistcodename"], Class['apt::update']] -> Package<| title == 'wireguard-dkms' |>
+      }
+      default : {
+        apt::pin {'wireguard':
+          packages => ['wireguard-dkms', 'wireguard-tools'],
+          release  => 'experimental',
+          priority => 501,
+          require  => Apt::Source['debian_unstable'],
```

Comment:
I think `Apt::Source['debian_unstable']` is defined somewhere else in internal puppet states. Will need to refactor this into a public module.